### PR TITLE
Regression template automl

### PR DESCRIPTION
**Pull request by prithvikannan**
## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

(Details)

We require adding a link to the [workflow](https://github.com/mlflow/mlflow/actions/workflows/recipe-template.yml) for testing the [mlflow/recipes-regression-template](https://github.com/mlflow/recipes-regression-template) code path.

Run the workflow with repository name: `mlflow/recipes-regression-template` and the branch`(Example:<user-name>:feature)` that you are working with.
### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/regression`: Sklearn regression recipe example
- [ ] `area/build`: Build and test infrastructure for MLflow recipe example

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: recipe.yaml [hunk 1]
```diff
@@ -43,9 +43,10 @@ steps:
     #                   applies input feature transformation during model training and inference.
     transformer_method: steps.transform.transformer_fn
   train:
-    using: estimator_spec
-    # Specifies the method that defines the estimator type and parameters to use for model training
-    estimator_method: steps.train.estimator_fn
```
Comment (sunishsheth2009) [inline recipe.yaml:48]: If we get rid of this, should we remove the code from `steps/train.py` since that is not needed anymore?

Comment (prithvikannan) [reply recipe.yaml:48]: I was thinking about this. I think it is important to show users how that code should be written (even if it is not used in this example). 

Specifically, we need users to return something with estimator params args such as `SGDRegressor(**estimator_params)` in order to pass in hardcoded and tuning params

WDYT? Does it make sense to make an "automl" example and a "estimator_spec" example? @sunishsheth2009

**Review by sunishsheth2009 — APPROVED** [review]
LGTM :) Thanks for making this change

File: steps/train.py [hunk 2]
```diff
@@ -16,6 +16,5 @@ def estimator_fn(estimator_params: Dict[str, Any] = {}):
     #
     # FIXME::OPTIONAL: return a scikit-learn-compatible regression estimator with fine-tuned
     #                  hyperparameters.
-    from sklearn.linear_model import SGDRegressor
 
-    return SGDRegressor(**estimator_params)
+    raise NotImplementedError
```
Comment (sunishsheth2009) [inline steps/train.py:20]: We might need to fix this test: https://github.com/mlflow/recipes-regression-template/blob/main/tests/train_test.py

**Review by sunishsheth2009 — CHANGES_REQUESTED** [review]

**Review by sunishsheth2009 — APPROVED** [review]

**Pull request merged**